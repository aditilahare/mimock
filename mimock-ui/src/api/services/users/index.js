import { getAllUsers } from './getUsers.service';
import { deleteUser } from './deleteUser.service';
import { getUserInfo } from './getUserInfo.service';
import { getUserRoles } from './getUserRoles.service';
import { updateUserActivationStatus } from './updateUserActivationStatus.service';
import { updateUserRole } from './updateUserRole.service';
import { updatePassword } from './updatePassword.service';
import { addNewUser } from './addUser.service';

export {
	getAllUsers,
	deleteUser,
	getUserInfo,
	getUserRoles,
	updatePassword,
	updateUserActivationStatus,
	updateUserRole,
	addNewUser,
};
