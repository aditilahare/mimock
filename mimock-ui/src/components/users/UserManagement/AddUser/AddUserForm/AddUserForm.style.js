import tw from 'tailwind-styled-components';
import { Button } from 'styles';

export const FormContainer = tw.form`
	w-11/12
	pl-32
`;

export const InputContainer = tw.div`
	flex
	items-center
	align-middle
	my-8
`;

export const InputBlock = tw.div`
	block
	h-12
	min-h-12
	max-h-12
`;

export const Label = tw.label`
	w-1/5
	font-sans
	font-bold
	subpixel-antialiased	
	text-lg
	text-gray-600
`;

export const FormInput = tw.input`
	block
	px-3
	py-1.5
	text-base
	font-normal
	text-gray-700
	bg-zinc-50
	bg-clip-padding
	border
	border-solid
	rounded
	mx-4
	focus:outline-blue-200
	${(props) => (props.$error ? 'border-2 border-rose-300' : 'border-gray-300')}
`;

export const UserRoleInput = tw.select`
    font-sans
    rounded-md
    shadow-md
    font-semibold
    p-2
    border-2
    border-gray-200
    outline-none
	mx-4
`;

export const ButtonContainer = tw.div`
  flex
  w-1/2
  my-20
  flex-row
  gap-4
`;

export const SubmitButton = tw(Button)``;

export const ResetButton = tw(Button)``;

export const ErrorMessage = tw.div`
	text-red-500
	text-sm
	font-sans
	subpixel-antialiased
	mt-1
	mb-3
	mx-4
	relative
`;

export const Error = tw.div`
    p-3
    my-10
    text-center
    bg-red-200
    border-2
    border-red-200
    mx-auto
    text-red-600
    font-sans
    w-3/4
    font-semibold
    rounded-lg
    text-lg
`;
