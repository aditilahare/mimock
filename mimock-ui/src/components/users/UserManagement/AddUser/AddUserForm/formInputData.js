export const formInputData = [
	{
		name: 'name',
		label: 'Name',
		type: 'text',
		placeholder: 'Enter name',
		validators: {
			required: 'Name is required',
			minLength: {
				value: 4,
				message: 'Name must be at least 6 characters',
			},
			maxLength: {
				value: 24,
				message: 'Name must be at most 24 characters',
			},
			pattern: {
				value: /^[a-zA-Z ]*$/,
				message: 'Name must be alphabetic',
			},
		},
	},
	{
		name: 'userName',
		label: 'Username',
		type: 'text',
		placeholder: 'Enter username',
		validators: {
			required: 'Username is required',
			minLength: {
				value: 4,
				message: 'Username must be at least 4 characters',
			},
			maxLength: {
				value: 24,
				message: 'Username must be at most 24 characters',
			},
		},
	},
	{
		name: 'password',
		nameConfirm: 'confirmPassword',
		label: 'Password',
		type: 'password',
		placeholder: 'Enter password',
		placeholderConfirm: 'Confirm password',
		validators: {
			required: 'Password is required',
			minLength: {
				value: 8,
				message: 'Password must be at least 8 characters',
			},
			maxLength: {
				value: 128,
				message: 'Password must be at most 128 characters',
			},
		},
	},
];
