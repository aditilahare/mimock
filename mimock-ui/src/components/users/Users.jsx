import React from 'react';
import UserManagement from './UserManagement';

export default function Users() {
	return <UserManagement />;
}
