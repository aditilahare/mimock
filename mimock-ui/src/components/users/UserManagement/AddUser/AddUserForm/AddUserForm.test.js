import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { getUserRoles, addNewUser } from 'services/users';
import AddUserForm from './AddUserForm';

const actualNotification = jest.requireActual('hooks/useNotification');

jest.mock('services/users');
jest.mock('react-notifications-component', () => {
	const Store = {
		addNotification: jest.fn(),
	};
	return { Store };
});

describe('AddUserForm', () => {
	beforeEach(() => {
		getUserRoles.mockResolvedValue([
			{
				roleName: 'ADMIN',
				roleDescription:
					'Administrator can create and manage mocks, and also has user management privileges',
			},
			{
				roleName: 'MANAGER',
				roleDescription:
					'Manager can create and manage mocks, but cannot manage users',
			},
			{ roleName: 'VIEWER', roleDescription: 'Viewer can only view mocks' },
		]);

		addNewUser.mockResolvedValue({});
	});

	it('should render add user form with all elements', async () => {
		let tree;
		await act(async () => {
			tree = await render(<AddUserForm />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(queryByTestId('user-role-error')).not.toBeInTheDocument();

		expect(getByTestId('add-user-form')).toBeInTheDocument();

		expect(getByTestId('input-name')).toBeInTheDocument();
		expect(getByTestId('input-userName')).toBeInTheDocument();
		expect(getByTestId('input-password')).toBeInTheDocument();
		expect(getByTestId('input-confirmPassword')).toBeInTheDocument();
		expect(getByTestId('input-role')).toBeInTheDocument();

		expect(queryByTestId('input-error-name')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-userName')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-password')).not.toBeInTheDocument();
		expect(
			queryByTestId('input-error-confirmPassword')
		).not.toBeInTheDocument();

		expect(getByTestId('add-user-submit-button')).toBeInTheDocument();
		expect(getByTestId('add-user-reset-button')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should show error banner when getUserRoles call fails', async () => {
		getUserRoles.mockRejectedValue(new Error('Error'));

		let tree;
		await act(async () => {
			tree = await render(<AddUserForm />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(getByTestId('user-role-error')).toBeInTheDocument();
		expect(getByTestId('user-role-error')).toHaveTextContent(
			'Failed to fetch user roles'
		);

		expect(queryByTestId('add-user-form')).not.toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should display validation errors when inputs are not valid', async () => {
		let tree;
		await act(async () => {
			tree = await render(<AddUserForm />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(queryByTestId('user-role-error')).not.toBeInTheDocument();

		expect(getByTestId('add-user-form')).toBeInTheDocument();

		expect(getByTestId('input-container-name')).toBeInTheDocument();
		expect(getByTestId('input-container-userName')).toBeInTheDocument();
		expect(getByTestId('input-container-password')).toBeInTheDocument();
		expect(getByTestId('input-container-role')).toBeInTheDocument();

		expect(getByTestId('input-name')).toBeInTheDocument();
		expect(getByTestId('input-userName')).toBeInTheDocument();
		expect(getByTestId('input-password')).toBeInTheDocument();
		expect(getByTestId('input-confirmPassword')).toBeInTheDocument();
		expect(getByTestId('input-role')).toBeInTheDocument();

		await act(async () => {
			fireEvent.click(getByTestId('add-user-submit-button'));
		});

		expect(getByTestId('input-error-name')).toBeInTheDocument();
		expect(getByTestId('input-error-userName')).toBeInTheDocument();
		expect(getByTestId('input-error-password')).toBeInTheDocument();

		expect(getByTestId('add-user-reset-button')).toBeInTheDocument();

		expect(addNewUser).not.toHaveBeenCalled();

		expect(container).toMatchSnapshot();
	});

	it('should call addNewUser api when all inputs are valid', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');

		let tree;
		await act(async () => {
			tree = await render(<AddUserForm />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(queryByTestId('user-role-error')).not.toBeInTheDocument();

		expect(getByTestId('add-user-form')).toBeInTheDocument();

		expect(getByTestId('input-container-name')).toBeInTheDocument();
		expect(getByTestId('input-container-userName')).toBeInTheDocument();
		expect(getByTestId('input-container-password')).toBeInTheDocument();
		expect(getByTestId('input-container-role')).toBeInTheDocument();

		expect(getByTestId('input-confirmPassword')).toBeInTheDocument();

		await act(async () => {
			fireEvent.change(getByTestId('input-name'), {
				target: { value: 'Tester Name' },
			});
			fireEvent.change(getByTestId('input-userName'), {
				target: { value: 'tester' },
			});
			fireEvent.change(getByTestId('input-password'), {
				target: { value: 'password' },
			});
			fireEvent.click(getByTestId('add-user-submit-button'));
		});

		expect(queryByTestId('input-error-name')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-userName')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-password')).not.toBeInTheDocument();
		expect(getByTestId('input-error-confirmPassword')).toBeInTheDocument();

		await act(async () => {
			fireEvent.change(getByTestId('input-confirmPassword'), {
				target: { value: 'password' },
			});
			fireEvent.change(getByTestId('input-role'), {
				target: { value: 'MANAGER' },
			});
			fireEvent.click(getByTestId('add-user-submit-button'));
		});

		expect(getByTestId('add-user-reset-button')).toBeInTheDocument();

		expect(addNewUser).toHaveBeenCalledTimes(1);
		expect(addNewUser).toHaveBeenCalledWith({
			name: 'Tester Name',
			userName: 'tester',
			password: 'password',
			userRole: 'MANAGER',
		});
		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
			message: 'User tester added successfully',
			title: 'New user added successfully',
			type: 'success',
		});

		expect(container).toMatchSnapshot();
	});

	it('should reset form on clicking reset button', async () => {
		let tree;
		await act(async () => {
			tree = await render(<AddUserForm />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(queryByTestId('user-role-error')).not.toBeInTheDocument();

		expect(getByTestId('add-user-form')).toBeInTheDocument();

		await act(async () => {
			fireEvent.change(getByTestId('input-name'), {
				target: { value: 'Tester Name' },
			});
			fireEvent.change(getByTestId('input-userName'), {
				target: { value: 'tester' },
			});
			fireEvent.change(getByTestId('input-password'), {
				target: { value: 'password' },
			});
			fireEvent.change(getByTestId('input-confirmPassword'), {
				target: { value: 'password' },
			});
		});

		expect(queryByTestId('input-error-name')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-userName')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-password')).not.toBeInTheDocument();
		expect(
			queryByTestId('input-error-confirmPassword')
		).not.toBeInTheDocument();

		await act(async () => {
			fireEvent.click(getByTestId('add-user-reset-button'));
		});

		expect(getByTestId('add-user-submit-button')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should show error notification when addNewUser api call fails', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		addNewUser.mockRejectedValueOnce(new Error('Error adding user'));

		let tree;
		await act(async () => {
			tree = await render(<AddUserForm />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(queryByTestId('user-role-error')).not.toBeInTheDocument();

		expect(getByTestId('add-user-form')).toBeInTheDocument();

		expect(getByTestId('input-container-name')).toBeInTheDocument();
		expect(getByTestId('input-container-userName')).toBeInTheDocument();
		expect(getByTestId('input-container-password')).toBeInTheDocument();
		expect(getByTestId('input-container-role')).toBeInTheDocument();

		expect(getByTestId('input-confirmPassword')).toBeInTheDocument();

		await act(async () => {
			fireEvent.change(getByTestId('input-name'), {
				target: { value: 'Tester Name' },
			});
			fireEvent.change(getByTestId('input-userName'), {
				target: { value: 'tester' },
			});
			fireEvent.change(getByTestId('input-password'), {
				target: { value: 'password' },
			});
			fireEvent.change(getByTestId('input-confirmPassword'), {
				target: { value: 'password' },
			});
			fireEvent.change(getByTestId('input-role'), {
				target: { value: 'MANAGER' },
			});
			fireEvent.click(getByTestId('add-user-submit-button'));
		});

		expect(queryByTestId('input-error-name')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-userName')).not.toBeInTheDocument();
		expect(queryByTestId('input-error-password')).not.toBeInTheDocument();
		expect(
			queryByTestId('input-error-confirmPassword')
		).not.toBeInTheDocument();

		expect(getByTestId('add-user-reset-button')).toBeInTheDocument();

		expect(addNewUser).toHaveBeenCalledTimes(1);
		expect(addNewUser).toHaveBeenCalledWith({
			name: 'Tester Name',
			userName: 'tester',
			password: 'password',
			userRole: 'MANAGER',
		});
		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
			message: 'Please try again',
			title: 'Failed to add new user',
			type: 'danger',
		});

		expect(container).toMatchSnapshot();
	});
});
