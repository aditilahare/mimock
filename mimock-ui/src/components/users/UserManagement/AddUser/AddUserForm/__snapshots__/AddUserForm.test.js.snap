// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserForm should call addNewUser api when all inputs are valid 1`] = `
<div>
  <form
    class="w-11/12 pl-32"
    data-testid="add-user-form"
  >
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-name"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Name
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-name"
          name="name"
          placeholder="Enter name"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-userName"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Username
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-userName"
          name="userName"
          placeholder="Enter username"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-password"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Password
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-role"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        User role
      </label>
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none mx-4"
        data-testid="input-role"
        type="select"
      >
        <option
          value="ADMIN"
        >
          ADMIN
        </option>
        <option
          value="MANAGER"
        >
          MANAGER
        </option>
        <option
          value="VIEWER"
        >
          VIEWER
        </option>
      </select>
    </div>
    <div
      class="flex w-1/2 my-20 flex-row gap-4"
    >
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-green-300 dark:bg-green-400"
        data-testid="add-user-submit-button"
      >
        Submit
      </button>
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-red-300 dark:bg-red-500"
        data-testid="add-user-reset-button"
      >
        Reset
      </button>
    </div>
  </form>
</div>
`;

exports[`AddUserForm should display validation errors when inputs are not valid 1`] = `
<div>
  <form
    class="w-11/12 pl-32"
    data-testid="add-user-form"
  >
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-name"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Name
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-2 border-rose-300"
          data-testid="input-name"
          name="name"
          placeholder="Enter name"
          type="text"
        />
        <div
          class="text-red-500 text-sm font-sans subpixel-antialiased mt-1 mb-3 mx-4 relative"
          data-testid="input-error-name"
        >
          Name is required
        </div>
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-userName"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Username
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-2 border-rose-300"
          data-testid="input-userName"
          name="userName"
          placeholder="Enter username"
          type="text"
        />
        <div
          class="text-red-500 text-sm font-sans subpixel-antialiased mt-1 mb-3 mx-4 relative"
          data-testid="input-error-userName"
        >
          Username is required
        </div>
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-password"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Password
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-2 border-rose-300"
          data-testid="input-password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
        <div
          class="text-red-500 text-sm font-sans subpixel-antialiased mt-1 mb-3 mx-4 relative"
          data-testid="input-error-password"
        >
          Password is required
        </div>
      </div>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-role"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        User role
      </label>
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none mx-4"
        data-testid="input-role"
        type="select"
      >
        <option
          value="ADMIN"
        >
          ADMIN
        </option>
        <option
          value="MANAGER"
        >
          MANAGER
        </option>
        <option
          value="VIEWER"
        >
          VIEWER
        </option>
      </select>
    </div>
    <div
      class="flex w-1/2 my-20 flex-row gap-4"
    >
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-green-300 dark:bg-green-400"
        data-testid="add-user-submit-button"
      >
        Submit
      </button>
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-red-300 dark:bg-red-500"
        data-testid="add-user-reset-button"
      >
        Reset
      </button>
    </div>
  </form>
</div>
`;

exports[`AddUserForm should render add user form with all elements 1`] = `
<div>
  <form
    class="w-11/12 pl-32"
    data-testid="add-user-form"
  >
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-name"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Name
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-name"
          name="name"
          placeholder="Enter name"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-userName"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Username
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-userName"
          name="userName"
          placeholder="Enter username"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-password"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Password
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-role"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        User role
      </label>
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none mx-4"
        data-testid="input-role"
        type="select"
      >
        <option
          value="ADMIN"
        >
          ADMIN
        </option>
        <option
          value="MANAGER"
        >
          MANAGER
        </option>
        <option
          value="VIEWER"
        >
          VIEWER
        </option>
      </select>
    </div>
    <div
      class="flex w-1/2 my-20 flex-row gap-4"
    >
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-green-300 dark:bg-green-400"
        data-testid="add-user-submit-button"
      >
        Submit
      </button>
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-red-300 dark:bg-red-500"
        data-testid="add-user-reset-button"
      >
        Reset
      </button>
    </div>
  </form>
</div>
`;

exports[`AddUserForm should reset form on clicking reset button 1`] = `
<div>
  <form
    class="w-11/12 pl-32"
    data-testid="add-user-form"
  >
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-name"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Name
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-name"
          name="name"
          placeholder="Enter name"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-userName"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Username
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-userName"
          name="userName"
          placeholder="Enter username"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-password"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Password
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-role"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        User role
      </label>
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none mx-4"
        data-testid="input-role"
        type="select"
      >
        <option
          value="ADMIN"
        >
          ADMIN
        </option>
        <option
          value="MANAGER"
        >
          MANAGER
        </option>
        <option
          value="VIEWER"
        >
          VIEWER
        </option>
      </select>
    </div>
    <div
      class="flex w-1/2 my-20 flex-row gap-4"
    >
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-green-300 dark:bg-green-400"
        data-testid="add-user-submit-button"
      >
        Submit
      </button>
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-red-300 dark:bg-red-500"
        data-testid="add-user-reset-button"
      >
        Reset
      </button>
    </div>
  </form>
</div>
`;

exports[`AddUserForm should show error banner when getUserRoles call fails 1`] = `
<div>
  <div
    class="p-3 my-10 text-center bg-red-200 border-2 border-red-200 mx-auto text-red-600 font-sans w-3/4 font-semibold rounded-lg text-lg"
    data-testid="user-role-error"
  >
    Failed to fetch user roles
  </div>
</div>
`;

exports[`AddUserForm should show error notification when addNewUser api call fails 1`] = `
<div>
  <form
    class="w-11/12 pl-32"
    data-testid="add-user-form"
  >
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-name"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Name
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-name"
          name="name"
          placeholder="Enter name"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-userName"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Username
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-userName"
          name="userName"
          placeholder="Enter username"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-password"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        Password
      </label>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div
        class="block h-12 min-h-12 max-h-12"
      >
        <input
          class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
          data-testid="input-confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
    </div>
    <div
      class="flex items-center align-middle my-8"
      data-testid="input-container-role"
    >
      <label
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      >
        User role
      </label>
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none mx-4"
        data-testid="input-role"
        type="select"
      >
        <option
          value="ADMIN"
        >
          ADMIN
        </option>
        <option
          value="MANAGER"
        >
          MANAGER
        </option>
        <option
          value="VIEWER"
        >
          VIEWER
        </option>
      </select>
    </div>
    <div
      class="flex w-1/2 my-20 flex-row gap-4"
    >
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-green-300 dark:bg-green-400"
        data-testid="add-user-submit-button"
      >
        Submit
      </button>
      <button
        class="px-9 py-3 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-red-300 dark:bg-red-500"
        data-testid="add-user-reset-button"
      >
        Reset
      </button>
    </div>
  </form>
</div>
`;
